libs/langchain/langchain/chains/combine_documents/reduce.py

```diff
@@ -43,9 +58,15 @@ def split_list_of_docs(
     """
     new_result_doc_list = []
     _sub_result_docs = []
+    _num_tokens = length_func([], **kwargs) if acum_length else 0
```

Comment:
why does it not start from `0`

Reply:
To give the oportunity of a custom init value. It is subtle, but you could use it to remove the length of the initial sepparator.

-sep + (sep + doc1) + (sep + doc2) + .... + (sep + docn)

the `length_func` could account for the document tokens plus some constant separator, and remove the initial separator for the empty array